Fix junos integration zuul CI failures

ganeshrn opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
* Fix netconf plugin to return `data_xml` object if present for the device handler object or else return raw xml

* Add check to run lldp integration test cases only if lldp is supported on the target host

Merged to devel https://github.com/ansible/ansible/pull/57309
(cherry picked from commit 0957835a4814bfc0c7649675963de5e88df5060e)
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- Test Pull request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
plugins/netconf/init.py
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

ganeshrn committed:
Fix junos integration zuul CI failures

(cherry picked from commit 0957835a4814bfc0c7649675963de5e88df5060e)

pabelanger commented:
recheck

abadger commented:
@ganeshrn The changelog entry doesn't seem to match with the PR.  Looking at the PR that went to devel, it looks like part of that commit isn't reflected here.  Perhaps, that code was left out on purpose but the changelog should be left out as well?

ganeshrn commented:
@abadger Thanks for pointing it out. I had left that intentionally out as the changes that went into devel branch to fix iosxr test case ended up breaking the junos tests.

I have updated the PR that should work with both iosxr and junos and also updated the changelog entry.

Please merge this PR only after the PR on devel https://github.com/ansible/ansible/pull/57981 is merged.

abadger commented:
#57981 merged to devel.  This one is now merged for the 2.8.2 release.
